### PR TITLE
`azurerm_monitor_diagnostic_setting` - `retention_policy` deprecated in favor of `azurerm_storage_management_policy`

### PR DESCRIPTION
**Pull request by teowa**
fix https://github.com/hashicorp/terraform-provider-azurerm/issues/23250

### CODE REVIEW

File: website/docs/r/monitor_diagnostic_setting.html.markdown [hunk 1]
```diff
@@ -159,6 +159,8 @@ A `retention_policy` block supports the following:
 * `enabled` - (Required) Is this Retention Policy enabled?
 
 * `days` - (Optional) The number of days for which this Retention Policy should apply.
+ 
+!> **NOTE:** `retention_policy` has been deprecated in favor of `azurerm_storage_management_policy` resource - to learn more information on the deprecation [in the Azure documentation](https://aka.ms/diagnostic_settings_log_retention).
```
Comment (tombuildsstuff) [inline website/docs/r/monitor_diagnostic_setting.html.markdown:163]: this should be moved up to line 157 and also copied under line 151

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
If we can fix up the documentation this should otherwise be good to go 👍

File: website/docs/r/monitor_diagnostic_setting.html.markdown (hunk 1 above)
Comment (teowa) [reply website/docs/r/monitor_diagnostic_setting.html.markdown]: fixed, thanks.

**Review by stephybun — APPROVED** [review]
Thanks @teowa LGTM 👍

**Comment by Panzerbjrn** [comment]
It would be really great to have some documentation to clearly show _how_ "azurerm_storage_management_policy" should be used in place of a simple 
```
    retention_policy {
      days    = 0 
      enabled = false
    }
```
block in azurerm_monitor_diagnostic_setting. Looking at [azurerm_storage_management_policy](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_management_policy) doesn't really help much.

**Comment by nkol2307** [comment]
any update on this ?

**Comment by Flo2388** [comment]
There is also something additional missing in my opinion in the documentation:

In the documentation for the `retention_policy` argument you can read:

> Setting this to 0 will retain the events indefinitely.

So our configuration looks like the following to disable the rule (enabled by default according to documentation) and set the storing to infinite:

```
  enabled_log {
    category = "logs"
    retention_policy {
      days    = 0
      enabled = false
    }
  }
```

What I can't find nor from the provider neither from the MS migration guide is, if the `retention_policy` block is deprecated, do I have to just remove the blocks to get infinite storing or do I have to implement the policy resource. And if the latter: What is the value to set for infinite logging? Just leave it to the defaults (-1), set it to 0 or what do I have to do?

Thanks for your help in advance.

**Pull request merged**